api/src/main/java/org/openmrs/validator/ConditionValidator.java

```diff
@@ -46,7 +48,9 @@ public void validate(Object object, Errors errors) {
 		Condition condition = (Condition) object;
 		if(condition.getCondition() == null){
 			errors.rejectValue("condition", "Condition.conditionShouldNotBeNull", "The condition is required");
-		}
+		} else if(condition.getClinicalStatus().equals(ConditionClinicalStatus.INACTIVE) && condition.getOnsetDate().compareTo(condition.getEndDate()) > 0){
```

Comment:
That'a detail but I personally find this a lot more readable using `Date.after(Date)`:
```java
else if (condition.getClinicalStatus().equals(ConditionClinicalStatus.INACTIVE) &&
  condition.getOnsetDate().after(condition.getEndDate())) {
```
You don't _have_ to change that, also this would push you in re-testing edge cases. But for you info.

Reply:
Thank you @mks-d . I will keep this in mind :)